src/completion/widget.ts

```diff
@@ -149,20 +149,24 @@ class CompletionWidget extends Widget {
    * Handle `after_attach` messages for the widget.
    *
    * #### Notes
-   * Captures document events in capture phase to dismiss or navigate the
+   * Captures window events in capture phase to dismiss or navigate the
    * completion widget.
+   *
+   * Because its parent (reference) widgets use window listeners instead of
+   * document listeners, the completion widget must also use window listeners
+   * in the capture phase.
```

Comment:
Even this may be insufficient, because it relies on the `window` listeners for the completion widget being added before the `window` listeners of the editor cell widget.

We may want to consider switching the editor cell widget to `document` listeners so that they are guaranteed to happen in lower priority than their parent widgets, since cells are always children of something else. @sccolbert may have some input on this, he's been thinking a lot about the life cycle and priority of keyboard shortcuts lately.

cc: @blink1073 @jasongrout